OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/19.3.0/setupGridEnv.sh

```diff
@@ -192,22 +192,43 @@ fi
 ####################################### Set Resetting Failed Units ##########################
 resetFailedUnits()
 {
-if [ "${RESET_FAILED_SYSTEMD}" != 'false' ]; then
-
-   cp $SCRIPT_DIR/$RESET_FAILED_UNITS  /var/tmp/$RESET_FAILED_UNITS
-   chmod 755 /var/tmp/$RESET_FAILED_UNITS
-   print_message "Setting Crontab"
-   cmd='su - $GRID_USER -c "sudo crontab $SCRIPT_DIR/$SET_CRONTAB"'
-   eval $cmd
-
-   if [ $?  -eq 0 ];then
-     print_message "Sucessfully installed $SET_CRONTAB using crontab"
-  else
-    error_exit "Error occurred in crontab setup"
-  fi
-
-fi
+        if [ "${RESET_FAILED_SYSTEMD}" != 'false' ]; then
+                # Define the service name
+                SERVICE_NAME="rhnsd"
+
+                # Check if the service is running
+                if pgrep -x "$SERVICE_NAME" >/dev/null; then
+                        print_message "$SERVICE_NAME is running."
+
+                        # Check if the service is responding
+                        if ! systemctl is-active --quiet "$SERVICE_NAME"; then
+                                print_message "$SERVICE_NAME is not responding. Stopping the service."
+                                systemctl stop "$SERVICE_NAME"
+                                systemctl disable "$SERVICE_NAME"
+                                print_message "$SERVICE_NAME stopped."
+                        else
+                                print_message "$SERVICE_NAME is responsive. No action needed."
+                        fi
+                else
+                        print_message "$SERVICE_NAME is not running."
+                fi
+
+        # shellcheck disable=SC2086
+        cp $SCRIPT_DIR/$RESET_FAILED_UNITS  /var/tmp/$RESET_FAILED_UNITS
```

Comment:
Is there a particular reason you don't just add quotes instead of disabling the check, i.e.

```shell
cp "$SCRIPT_DIR/$RESET_FAILED_UNITS"  "/var/tmp/$RESET_FAILED_UNITS"
```

Reply:
Well, I am OK to quote it, but changing it in last moment for me means I have to redo all tests of what i tested in the weekend and this is urgent customer escalation and we need this to be merged asap. These checks are only detected during upload to github. At the moment, I am not aware of how to selectively run "shellcheck" of changed/delta code only which github is running. If you insist, i can quote it and run one simple test. FYI, we do have improvements plans to merge this week. By today is priority for merge.

Reply:
https://www.shellcheck.net/ can be installed on Windows, macOS and Linux and is available as a plugin for Visual Studio Code as well as all the JetBrains IDEs, as are all the enabled linters which are listed in the contribution guide.

There is also the `lint` script in the root of this repo that will run the checks for you locally. 

Note that this repo is also not a production service and there is no SLA for merging updates. The folks who review and maintain it are doing so in addition to their actual jobs, not as part of it.